File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution{ 
public:
    int majorityElement(vector<int>& nums) {


        // time O(n) space O(n)
        // unordered_map<int,int> mp;
        // for(int i:nums){
        //     mp[i]++;
        // }

        // int ans=0;
        // int index ;
        // for(auto it:mp)
        // {
        //     if(it.second>ans) {
        //         ans=it.second;
        //         index = it.first;
        // }
        // }
        // return index;


        int candidate = 0 ; 
        int count = 0 ;

        for(int x : nums)
        {
            if(count==0)
            {
                candidate=x;
            }

            if(x==candidate)
            {
                count++;
            }
            else count--;
        }
        return candidate;

        
    }
};

File: Data Structures & Algorithms/is-anagram/submission-1.java
class Solution {
    public boolean isAnagram(String s, String t) {
          int ans[] = new int[26];

          if(s.length()!=t.length())return false;

          for(int i = 0 ; i<s.length() ; i++){
            ans[s.charAt(i)-'a']++;
            ans[t.charAt(i)-'a']--;
            }

        for (int x : ans) {
            if (x != 0) return false;
        }
        return true;
          

        
    }
}
